### DISCUSSION TITLE
how about deny some plugins config when APISIX-OpenResty is missed?

### DISCUSSION THREAD

**Discussion by RocFang** [apache/apisix#4964] [Ideas]
Some plugins need APISIX-OpenResty, for example client-contol etc.

currently, user can still enable them, and retun 501 when access.

Maybe it's more friendly to deny the config request at first like jsonschema?

 for example,  add a local bolean module-level viable in these plugins, and check it right before of after check_schema.

**Comment by spacewander** [comment]
Deny the config during checking the schema may bring inconvenience when we are doing A/B testing.
